Library/Homebrew/cask/audit.rb

```diff
@@ -270,6 +271,16 @@ def bad_osdn_url?
       bad_url_format?(/osd/, [%r{\Ahttps?://([^/]+.)?dl\.osdn\.jp/}])
     end
 
+    def check_url_verification
+      return unless cask.url.verified
```

Comment:
This will make it succeed if there is no `verified:` parameter, but that’s not the goal of the check. The point is to detect *when it’s needed but it’s not present*, and if it’s correct when present.

It should:

* Check if `url` and and `homepage` have the same hostname.
* If they do, we can `return`.
* If they don’t, check if `url` and `verified` match.
* If they do, we can `return`.
* If they don’t, throw the error and point to the documentation.

It might be useful to consult the `style` check and see how it’s done there. That check should also be removed in this PR.

In addition, it would be helpful if some common hosts had longe checks. For example, a Github repo download follows the form `https://github.com/user/repo`. In those cases, we should check that much (not just `github.com`) before marking as correct. That said, that can be done afterwards if you don’t want to pursue it that far. But for the moment this PR should make the `audit` at least as good as the `style` check.